exercises/dot-dsl/README.md

```diff
@@ -0,0 +1,52 @@
+# Dot Dsl
+
+Write a Domain Specific Language similar to the Graphviz dot language.
+
+A [Domain Specific Language
+(DSL)](https://en.wikipedia.org/wiki/Domain-specific_language) is a
+small language optimized for a specific domain.
+
+For example the dot language of [Graphviz](http://graphviz.org) allows
+you to write a textual description of a graph which is then transformed
+into a picture by one of the graphviz tools (such as `dot`). A simple
+graph looks like this:
+
+    graph {
+        graph [bgcolor="yellow"]
+        a [color="red"]
+        b [color="blue"]
+        a -- b [color="green"]
+    }
+
+Putting this in a file `example.dot` and running `dot example.dot -T png
+-o example.png` creates an image `example.png` with red and blue circle
+connected by a green line on a yellow background.
+
+Create a DSL similar to the dot language.
+
+## Description of DSL
+
+A graph, in this DSL, is an object of type `Graph`, taking a list of one 
+or more
+
++ attributes
++ nodes
++ edges
+
+described as tuples.
+
+The implementations of `Node` and `Edge` provided in `dot_dsl.py`.
+
+Observe the test cases in `dot_dsl_test.py` to understand the DSL's design.
+
+### Submitting Exercises
```

Comment:
Per #950 , this should be at heading level 2 instead of 3.

Reply:
Moved it to level 2!